libs/domain/user/address-list/address-list.component.spec.ts

```diff
@@ -28,96 +61,225 @@ describe('UserAddressListComponent', () => {
-        (
-          element.renderRoot.querySelector(
-            'oryx-tile:nth-child(2) input'
-          ) as HTMLInputElement
-        ).dispatchEvent(new InputEvent('input'));
-
-        element.requestUpdate();
-        await elementUpdated(element);
+        element = await fixture(
+          html`<oryx-user-address-list
+            .options=${{ selectable: true }}
+          ></oryx-user-address-list>`
+        );
+      });
+
+      it('should render an input for the items', () => {
+        expect(element).toContainElement(`input[value='billing']`);
+        expect(element).toContainElement(`input[value='shipping']`);
+      });
+
+      it('should not select any inputs', () => {
+        expect(element).not.toContainElement('input[checked]');
+      });
+
+      describe('and when an address is selected', () => {
+        beforeEach(async () => {
+          addressStateService.get.mockReturnValue(of({ selected: 'shipping' }));
+          element = await fixture(
+            html`<oryx-user-address-list
+              .options=${{ selectable: true }}
+            ></oryx-user-address-list>`
+          );
+        });
+
+        it('should select the tile', () => {
+          expect(element).toContainElement(
+            `oryx-tile[selected] input[value='shipping']`
+          );
+        });
+      });
+
+      describe('and when an address is selected', () => {
+        beforeEach(() => {
+          element.dispatchEvent = vi.fn();
+          element.renderRoot
+            .querySelector<HTMLInputElement>(`input[value='shipping']`)
+            ?.click();
+        });
+
+        it('should set the selected item in the address state service', () => {
+          expect(addressStateService.set).toHaveBeenCalled();
+        });
+
+        it('should dispatch a custom change event', () => {
+          expect(element.dispatchEvent).toHaveBeenCalledWith(
+            new CustomEvent('change', {
+              bubbles: true,
+              composed: true,
+              detail: { address: mockBillingAddress },
+            })
+          );
+        });
       });
+    });
+
+    describe('when the address state = Update', () => {
+      beforeEach(async () => {
+        addressStateService.get.mockReturnValue(
+          of({ action: CrudState.Update })
+        );
+      });
+
+      describe('and the editTarget is modal', () => {
+        beforeEach(async () => {
+          element = await fixture(
+            html`<oryx-user-address-list
+              .options=${{
+                editTarget: EditTarget.Modal,
+              } as UserAddressEditComponentOptions}
+            ></oryx-user-address-list>`
+          );
+        });
+
+        it('should show the modal', () => {
+          expect(element).toContainElement('oryx-modal');
+        });
+
+        describe('and the close event is dispatched', () => {
+          const event = new CustomEvent<AddressEventDetail>('oryx.close');
+          event.stopPropagation = vi.fn();
```

Comment:
```suggestion
          spy = vi.spyOn(event, 'stopPropagation');
```

and 

expect(spy).toHaveBeenCalled...